cmd/serviceusers/create.go

```diff
@@ -0,0 +1,176 @@
+package serviceusers
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/appgate/sdp-api-client-go/api/v17/openapi"
+	"github.com/appgate/sdpctl/pkg/docs"
+	"github.com/appgate/sdpctl/pkg/factory"
+	"github.com/appgate/sdpctl/pkg/filesystem"
+	"github.com/appgate/sdpctl/pkg/prompt"
+	"github.com/appgate/sdpctl/pkg/util"
+	"github.com/hashicorp/go-multierror"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+func NewServiceUsersCreateCMD(f *factory.Factory) *cobra.Command {
+	opts := ServiceUsersOptions{
+		Config: f.Config,
+		API:    f.ServiceUsers,
+		In:     f.Stdin,
+		Out:    f.IOOutWriter,
+	}
+	cmd := &cobra.Command{
+		Use:     "create",
+		Short:   docs.ServiceUsersCreate.Short,
+		Long:    docs.ServiceUsersCreate.Long,
+		Example: docs.ServiceUsersCreate.ExampleString(),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if opts.Config.Version <= 16 {
+				return fmt.Errorf("The service user interface is only available from API version 17 or higher. Currently using API version %d", opts.Config.Version)
+			}
```

Comment:
I think it would be enough to append a `PersistentPreRunE` func in `service-users` (serviceusers.go) instead of doing this 5 times in every subcommand.

Reply:
That would overwrite the `PersistentPreRunE` from root though, which is kind of important. That's why it's repeated.

Reply:
So instead, I implemented a version restriction using command annotation instead and do the actual checking in the root `PersistentPreRunE`. I think it may be useful to have for cases like these.